Code/Editor/EditorFramework/Assets/Implementation/AssetCurator.cpp

```diff
@@ -1603,89 +1619,10 @@ void ezAssetCurator::HandleSingleFile(const ezString& sAbsolutePath)
         }
       }
     }
-
-    return;
-  }
-  else
-  {
-    EZ_ASSERT_DEV(!Stats.m_bIsDirectory, "Directories are handled by ezAssetWatcher and should not pass into this function.");
-  }
-
-  HandleSingleFile(sAbsolutePath, Stats);
-}
-
-void ezAssetCurator::HandleSingleFile(const ezString& sAbsolutePath, const ezFileStats& FileStat)
-{
-  EZ_ASSERT_DEV(!FileStat.m_bIsDirectory, "Directories are handled by ezAssetWatcher and should not pass into this function.");
-  CURATOR_PROFILE("HandleSingleFile2");
-  EZ_LOCK(m_CuratorMutex);
-
-  ezStringBuilder sExt = ezPathUtils::GetFileExtension(sAbsolutePath);
-  sExt.ToLower();
-
-  // store information for every file, even when it is no asset, it might be a dependency for some asset
-  auto& RefFile = m_ReferencedFiles[sAbsolutePath];
-
-  // mark the file as valid (i.e. we saw it on disk, so it hasn't been deleted or such)
-  RefFile.m_Status = ezFileStatus::Status::Valid;
-
-  bool fileChanged = !RefFile.m_Timestamp.Compare(FileStat.m_LastModificationTime, ezTimestamp::CompareMode::Identical);
-  if (fileChanged)
-  {
-    RefFile.m_Timestamp.Invalidate();
-    RefFile.m_uiHash = 0;
-    if (RefFile.m_AssetGuid.IsValid())
-      InvalidateAssetTransformState(RefFile.m_AssetGuid);
-
-    auto it = m_InverseTransformDeps.Find(sAbsolutePath);
-    if (it.IsValid())
-    {
-      for (const ezUuid& guid : it.Value())
-      {
-        InvalidateAssetTransformState(guid);
-      }
-    }
-
-    auto it2 = m_InverseThumbnailDeps.Find(sAbsolutePath);
-    if (it2.IsValid())
-    {
-      for (const ezUuid& guid : it2.Value())
-      {
-        InvalidateAssetTransformState(guid);
-      }
-    }
-  }
-
-  // Assets should never be in an AssetCache folder.
-  const char* szNeedle = sAbsolutePath.FindSubString("AssetCache/");
-  if (szNeedle != nullptr && sAbsolutePath.GetData() != szNeedle && szNeedle[-1] == '/')
-  {
-    return;
-  }
-
-  // check that this is an asset type that we know
-  if (!m_ValidAssetExtensions.Contains(sExt))
-  {
-    return;
-  }
-
-  // the file is a known asset type
-  // so make sure it gets a valid GUID assigned
-
-  // File hasn't change, early out.
-  if (RefFile.m_AssetGuid.IsValid() && !fileChanged)
-    return;
-
-  // store the folder of the asset
-  {
-    ezStringBuilder sAssetFolder = sAbsolutePath;
-    sAssetFolder = sAssetFolder.GetFileDirectory();
-
-    m_AssetFolders.Insert(sAssetFolder);
+    break;
+    default:
```

Comment:
should we just ignore all cases, or assert that we covered all known cases ?

Reply:
Added assert